programs/platform.h

```diff
@@ -0,0 +1,130 @@
+/*
+    platform.h - compiler and OS detection
+    Copyright (C) 2016-present, Przemyslaw Skibinski, Yann Collet
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
```

Comment:
We are supposed to use [BSD license](https://github.com/facebook/zstd/blob/dev/LICENSE) for Facebook Open-Source projects.
There can be exceptions, sure, but for a file which belongs to the core cli experience, it's likely not possible nor desirable to make such an exception.